dockertest/docker_daemon.py

```diff
@@ -112,51 +111,136 @@ def version(self):
 # Group of utils for managing docker daemon service.
 
 
-def start(docker_path, docker_args):
+def _which_docker():
     """
-    Start new docker daemon with special args.
+    Returns 'docker' or 'docker-latest' based on setting in
+    /etc/sysconfig/docker.
 
-    :param docker_path: Full path to executable
-    :param docker_args: List of string of command-line arguments to pass
-    :returns: Opaque daemon_process object (not for direct use)
+    Warning: this is not a reliable method. /etc/sysconfig/docker defines
+    the docker *client*; it is perfectly possible for a system to use
+    docker as client and docker-latest as daemon or vice-versa. It's
+    possible, but unsupported, so we're not going to worry about it.
     """
-    # _SpecificServiceManager creates it's methods during __init__()
-    if service.get_name_of_init() == "systemd":
-        # pylint: disable=E1101
-        utils.run("systemctl stop docker.socket", ignore_status=True)
-    service.SpecificServiceManager("docker").stop()  # pylint: disable=E1101
-    cmd = [docker_path]
-    cmd += docker_args
+    docker = 'docker'
+    with open('/etc/sysconfig/docker', 'r') as docker_sysconfig:
+        for line in docker_sysconfig:
+            if line.startswith('DOCKERBINARY='):
+                if 'docker-latest' in line:
+                    docker = 'docker-latest'
+    return docker
 
-    daemon_process = utils.AsyncJob(" ".join(cmd), close_fds=True)
-    return daemon_process
 
+def _systemd_action(action):
+    utils.run("systemctl %s %s.service" % (action, _which_docker()))
+    # FIXME: check status
```

Comment:
Interesting, so #1 must depend on both the unit's service-type and what the failure is.  I've been playing around with openvpn (totally different context) and it uses ``Type=forking`` (whereas docker uses Type=notify``.

I just tried it to confirm- If I screw up openvpn's config file, then ``systemctl start openvpn@server.service`` exits 0.  However, if I then run ``systemctl status openvpn@server.service`` it shows it as failed.  But I bet if I screwed up the unit file itself, then the ``start`` would exit() != 0.

Don't you just love how working with systemd is **sooooooooo** simple!

Though, even if at some future point docker's startup type changes. Won't the tests fail anyway if the daemon isn't actually running after ``docker_daemon.start()`` is called?  So maybe this is good enough.